### DISCUSSION TITLE
Office365 reading messages slow since the weekend 2024-05-12|13 - anybody else having the same issue?

### DISCUSSION THREAD

**Discussion by rogatec** [jstedfast/MailKit#1752] [General]
Hi everyone,

I'am having trouble reading messages via `await folder.GetMessageAsync(id, _token);` since Sunday/Monday and each day is getting worse.

I am crawling a mailbox with currently 2200 mails:
```csharp
await folder.OpenAsync(FolderAccess.ReadWrite, _token);
var uidList = await folder.SearchAsync(SearchQuery.All, _token);
```

After that I am iterating over the list and call the `GetMessageAsync` - until last weekend this process took no longer than 60 seconds.

But then things getting weird, because now I often get an exception:
```csharp
Operation timed out after 120000 milliseconds
System.IO.IOException: The read operation failed, see inner exception.
 ---> System.TimeoutException: Operation timed out after 120000 milliseconds
 ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
```

As you can see the `mailClient` timeout is set to 120 seconds - which worked fine until the weekend.

Now the same logic takes between sometimes 10minutes up to almost 50minutes.

I had the chance to debug the mailbox locally and it is brutally slow.

Opening the folder and searchAll does take a fraction of a second, but **one** `getMessageAsync` call with an uid took 26.698ms.

Are there any news out where I missed something?

**Comment by jstedfast** [comment]
Are you downloading every single message every single day? You probably hit some sort of throttling situation that is meant to defend against DoS attacks.

**Reply by rogatec** [reply]
Hm yes maybe... but it was running on different systems with different O365 connections for months (some for a year)...

But yeah I already changed the logic in a newer version.
But it is very suspicious because 3 customers started complaining yesterday and today and the internal system is also struggling.
